s4e-backend/src/main/java/pl/cyfronet/s4e/ex/ErrorHandlerHelper.java

```diff
@@ -52,6 +52,30 @@
         return map;
     }
 
+    public Map<String, Object> toResponseMap(QueryException e) {
+        BindingResult bindingResult = e.getBindingResult();
+        val map = new LinkedHashMap<String, Object>();
+
+        // populate general errors
+        if (bindingResult.hasGlobalErrors()) {
+            map.put("__general__", bindingResult.getGlobalErrors().stream()
+                    .map(objectError -> getMessage(objectError))
+                    .collect(Collectors.toUnmodifiableList()));
+        }
```

Comment:
Could you move this if to function `populateGeneralErrors`?

Reply:
I would keep it there, as it's not so complicated, and there are only two places it's used in, see "rule of three"

Reply:
The root of the idea is: put blocks of code in more contextual places - methods - (instead of adding comments like `populate general errors` above them). In this case, the comment wasn't caused by duplication. 

In general: `abstraction should go from top to down`, where detailed method name should describe what they're exactly doing inside. Those methods with comments are a perfect example of it.

Reply:
I understand your argument, that one could use language constructs to describe a process. However, I see these constructs also as a hindrance, since you have to switch context to see exactly what is done in a separate method which may be in a different region of the class. Also, making the code conform to some specific requirement may be more difficult, as when you start to reuse the extracted method you have to care about its abstraction.

I agree that _abstraction should go from up to down_, however, it doesn't require you to extract all the tiniest possible abstraction. Instead, weigh pros and cons, in this case I think that level was enough.

Reply:
I know what you mean saying that there will be `context switch`, but as long as a method is well named (in this case it is) we don't need to dive into the function to understand what it is doing. Let's say that this case isn't `cornerstone` of the PR to discuss it any longer, it is fine as it is. 

PS For the future, mostly I give `approval` with comments that are nice to have for me. Most of them are related to style or placement. If you don't agree with them, just skip it and merge the PR, nothing wrong will happen :)